string name
int64 num
